### PR TITLE
Fixes #276: Add config stack

### LINKED ISSUE #276: CI job "py27-pytestlatest-xdist" fails
The job is failing lately, and this can be reproduced (without using xdist) using this particular schedule:
```
$ tests/feature/test_no_sctrict_gherkin.py::test_background_no_strict_gherkin  tests/feature/test_wrong.py::test_verbose_output
```

It seems that the setup of `test_background_no_strict_gherkin` leaks into `test_verbose_output`. I think this is a pytest issue and I am trying to create a smaller file to reproduce the issue.

### PR DESCRIPTION
**Pull request by sliwinski-milosz**
Simplified solution for failing tests.

Prepared based on comments from https://github.com/pytest-dev/pytest-bdd/pull/282

I also added two small fixes for `pytest==4.2.0`

### CODE REVIEW

File: pytest_bdd/gherkin_terminal_reporter.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def pytest_runtest_logstart(self, nodeid, location):
 
     def pytest_runtest_logreport(self, report):
         rep = report
-        res = self.config.hook.pytest_report_teststatus(report=rep)
+        res = self.config.hook.pytest_report_teststatus(report=rep, config=self.config)
```
Comment (sliwinski-milosz) [inline pytest_bdd/gherkin_terminal_reporter.py]: `output_must_be_the_same_as_regular_reporter` check started to fail due to below warning on pytest 4.2.0:
```
test_gherkin.py::test_scenario_1
  /Users/milosz.sliwinski/workspace/pytest-bdd/.tox/py27-pytestlatest/lib/python2.7/site-packages/pytest_bdd/gherkin_terminal_reporter.py:61: UserWarning: Argument(s) ('config',) which are declared in the hookspec can not be found in this hook call
    res = self.config.hook.pytest_report_teststatus(report=rep)
```
Indeed [pytest_report_teststatus hook](https://docs.pytest.org/en/latest/reference.html#_pytest.hookspec.pytest_report_teststatus) takes `config` as an argument so I just passed it and warning disappeared.

File: tests/feature/test_tags.py [hunk 2]
```diff
@@ -127,7 +127,7 @@ def i_have_bar():
         scenarios('test.feature')
     """)
     result = testdir.runpytest('-rsx')
-    result.stdout.fnmatch_lines(["SKIP *: Not implemented yet"])
+    result.stdout.fnmatch_lines(["SKIP*: Not implemented yet"])
```
Comment (sliwinski-milosz) [inline tests/feature/test_tags.py]: On `pytest==4.1.1` message in `short test summary info` is:
```
SKIP [1] test_apply_tag_hook.py:40: Not implemented yet
```

But on `pytest==4.2.0` it is:
```
SKIPPED [1] test_apply_tag_hook.py:40: Not implemented yet
```

I fixed above test to work with both messages.

**Comment by youtux** [comment]
Looks good to me too, good job!

**Comment by The-Compiler** [comment]
So looks like things are working normally again? Probably time for a new release then? @youtux if you have the time to cut a new release, that'd be much appreciated - if not, I can try doing one, but it's been a while since I last did.

**Comment by youtux** [comment]
The release would just be for the warning being fixed in https://github.com/pytest-dev/pytest-bdd/pull/288#discussion_r255415594, right?

**Comment by The-Compiler** [comment]
It breaks the entire pytest run when treating warnings as errors (which IMHO is a good thing to do in testsuites). However, I suppose I could ignore that warning in my project in the meantime.

**Comment by youtux** [comment]
I'll make a release, no problems

**Comment by youtux** [comment]
I just release version 3.0.2. Happy testing!

**Pull request merged**